### ISSUE TITLE
Error when extending `javax.swing.JTable` and `javax.swing.text.JTextComponent`

### ISSUE THREAD

**Issue by EotT123** [manifold-systems/manifold#650]
**Describe the bug**
When extending certain javax.swing components, such as `javax.swing.JTable` and `javax.swing.text.JTextComponent`, compilation fails with the following error message:

```
[ERROR] javax/swing/text/JTextComponent.java:[407,22] error: constructor DropLocation in class DropLocation cannot be applied to given types;
[ERROR]   required: Point
[ERROR]   found:    no arguments
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] javax/swing/JTable.java:[566,22] error: constructor DropLocation in class DropLocation cannot be applied to given types;
[ERROR]   required: Point
[ERROR]   found:    no arguments
[ERROR]   reason: actual and formal argument lists differ in length
```

This issue occurs when creating an extension class for these classes, even without adding any extension methods. For example:

```java
@Extension
public class JTextComponentExt {
}
```

It seems that this error is specific to some `javax.swing` components, as many others do not trigger the issue. 

**Desktop (please complete the following information):**
- OS Type & Version: Windows 10 22H2
- Java/JDK version: openjdk 23.0.1
- IDE version (IntelliJ IDEA or Android Studio): IntelliJ IDEA 2024.3.1.1
- Manifold version: 2024.1.43 (latest version)
- Manifold IntelliJ plugin version: 2024.1.13 (latest version)

**Comment by rsmckinney** [comment]
Interesting. . .

tl;dr  a fix is complete and will be available in the next release, coming soon


*For the record*

This has to do with targeting an older JDK than the compiling JDK, which I presume your project is doing? For instance, compiling with JDK 21 and targeting JDK 17 can trigger this issue.

The problem is caused by _how_ a JDK is able to target an older JDK's API without bundling all prior JDK APIs. In a nutshell rather than bundling every major JDK API release, type information diffs are maintained in a simple format in the JDK's lib/ct.sym file, which the compiler uses to build JDK class symbols as if loaded from a class file. However, to keep the ct.sym file nice and tight only _accessible_ type info is recorded since, under normal circumstances, private stuff should never be directly accessed. And, normally, that works out nicely, normally. But when generating type stubs, which is part of what manifold extension code gen does, there is an issue--this issue--dealing with private constructors when the declaring class has a super class :/

**Comment by EotT123** [comment]
Thank you for the quick response! 
I'm using and targeting JDK 23, so there might be another cause for this issue. I'll test it once the fix is released.

**Comment by EotT123** [comment]
With Manifold 2024.1.44, I'm encountering a compilation error again, but this time without any useful information.

However, I found a solution based on your conclusion. In my `pom.xml`, I had the following configuration:
```xml
<properties>
    <maven.compiler.target>23</maven.compiler.target>
    <maven.compiler.source>23</maven.compiler.source>
    <maven.compiler.release>23</maven.compiler.release>
<properties>
```

By removing the `maven.compiler.release` property, it works correctly (even with 2024.1.43).

**Comment by rsmckinney** [comment]
I think you can also just use the release property:
```xml
<properties>
    <maven.compiler.release>23</maven.compiler.release>
<properties>
```
The difference is subtle, but addresses the case where your intention is to run on Java 23+ when compiling with JDK 24+. Essentially, a 24+ compiler won't let your code reference 24+ specific stuff. Shrug.
